examples/optuna/skorch_example.py

```diff
@@ -0,0 +1,124 @@
+import argparse
+import urllib
+
+import numpy as np
+import optuna
+from optuna.integration import SkorchPruningCallback
+import skorch
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import pandas as pd
+
+from sklearn.datasets import fetch_openml
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import train_test_split
+
+
+SUBSET_RATIO = 0.4
+
+mnist = fetch_openml("mnist_784", cache=False)
+
+X = pd.DataFrame(mnist.data)
+y = mnist.target.astype("int64")
+indices = np.random.permutation(len(X))
+N = int(len(X) * SUBSET_RATIO)
+X = X.iloc[indices][:N].astype(np.float32)  
+y = y[indices][:N]
+
+X /= 255.0
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=42)
+device = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+class ClassifierModule(nn.Module):
+    def __init__(self, n_layers: int, dropout: float, hidden_units: list[int]) -> None:
+        super().__init__()
+
+        layers = []
+        input_dim = 28 * 28  # Assuming flattened MNIST input
+
+        for i in range(n_layers):
+            layers.append(nn.Linear(input_dim, hidden_units[i]))
+            layers.append(nn.Dropout(dropout))
+            layers.append(nn.ReLU())
+            input_dim = hidden_units[i]
+
+        layers.append(nn.Linear(input_dim, 10))
+        layers.append(nn.Softmax(dim=-1))  # Final softmax layer
+
+        self.model = nn.Sequential(*layers)
+
+    def forward(self, x):
+        if isinstance(x, dict):
+            x = x["data"]
+        return self.model(x)
+
+X_train_np = X_train.to_numpy().astype(np.float32)
+X_test_np = X_test.to_numpy().astype(np.float32)
+y_test_np = y_test.to_numpy()
+
+def objective(trial: optuna.Trial) -> float:
+    # Define hyperparameter search space
+    n_layers = trial.suggest_int("n_layers", 1, 3)
+    dropout = trial.suggest_float("dropout", 0.2, 0.5)
```

Comment:
Let's allow the value to go to 0.

```suggestion
    dropout = trial.suggest_float("dropout", 0.0, 0.5)
```

Reply:
Updated as discussed.